File: 1.14/main.py
def find_majority(v):
    count=0;
    maj=0;
    for i in v:
        if count == 0:
            maj=i
            count=1
        elif maj==i:
            count+=1
        else:
            count-=1
    return maj

import random

v = [random.randint(0,9) for r in xrange(100)]
print v
print "maj = ", find_majority(v)

File: 1.14/main.cpp
#include <vector>
#include <iostream>
#include <algorithm>
#include <iterator>
#include <string>
#include <random>

using namespace std;

template <typename t>
typename t::value_type find_majority(t const &v) {
  typename std::remove_reference<decltype(v)>::type::value_type candidate;
  int count=0;
  for(auto i:v){
    if(count==0) {
      candidate=i;
      count =1;
    } else if (candidate==i) count++;
    else count--;
  }
  return candidate;
}

int main() {
  random_device rd;
  mt19937 g(rd());
  uniform_int_distribution<> d(0,9);
  vector<int> v(100);
  generate( begin(v), end(v),[&](){return d(g);});
  std::copy( std::begin(v), std::end(v),std::ostream_iterator<decltype(v)::value_type>(cout," ")); cout << endl;
  cout << "find majority = " << find_majority(v) << endl;
  return 0;
}
